tutorials/migration.md

```diff
@@ -4,96 +4,846 @@ layout: intro
+cp <openHAB 1.x conf>/configurations/rules/* <openHAB 2 conf>/rules/*
+cp <openHAB 1.x conf>/configurations/sitemaps/* <openHAB 2 conf>/sitemaps/*
+~~~~
+
+##Necessary Changes
+
+
+###Items
+
+As mentioned above, you must now manually install any Transformation engine your
+Items may use.
+
+The **SCALE** transformation has evolved.
+ 
+- Old syntax that was `[minbound,maxbound]` has to be changed to 
+`[minbound..maxbound]`. 
+- Note that you now have the ability to exclude bounds from the ranges (e.g. 
+`]minbound..maxbound]`) and also define open ranges`[minbound..]`
+
+
+###Sitemap
+
+If you use png icons, you must change the default icons from svg to png for
+ClassicUI and BasicUI. This can be done in PaperUI in `Configuration -> Service->BasicUI and ClassicUI`.
+Set the "Default Icon Format" to "Bitmap".
+
+Note that not all of the default icons that came with openHAB 1.x are avaialble 
+in the default set for openHAB 2. If you are missing an icon in your sitemap 
+that could be the cause. The fill list of openHAB 2 icons is [here](../addons/iconsets/classic/readme.html).
+
+Also, in openHAB 2 there is a new requirement for dynamic icons: there must be a 
+default. For example, if one has a bunch of Wunderground icons (e.g. 
+wunderground-chanceflurries.png) there mst be a `wunderground.png` icon as well.
+
+Note that the path to the Sitemap UIs have changed. If one goes
+to `http://<hostname>:8080` you will see a list of all the installed UIs, 
+including the administration UIs. To get to ClassicUI the URL is
+`http://<hostname>:8080/classicui/app?sitemap=<your sitemap>`. If you previously
+used `default.sitemap` as your file name so you could skip the `?sitemap=` part
+you must now rename the file to `_default.sitemap` and change the very first
+line of the file to read `sitemap _default label="<sitemap name>`. 
+
+openHAB 2 only supports static webviews. The dynamic webview some 
+created with the Weather Binding in openHAB 1.x are not supported. Habpanel is an 
+excellent alternative approach.
+
+###Rules
+One set of errors that will occur in rules that refer to the packages of openHAB
+core classes (e.g. org.openhabs.core.*). These classes have moved to 
+org.eclipse.smarthome.core.*. Furthermore, these classes as well as the Joda 
+time classes no longer need to be imported.
+
+In addition, to test an Item to see if it is Undefined one tests against `NULL`
+instead of `Undefined`. For example `if(MyItem.state == Undefined)` becomes
+`if(MyItem.state == NULL)`. Note that `NULL` is not to be confused with `null`.
+`NULL` is only valid when testing an Item's state to see if it is undefined.
+`null` is used pretty much everywhere else to mean "no value" and usually 
+indicates no result or an error.
+
+Finally, the `HSBType` state type can no longer be constructed using a `java.awt.Color` object, and there is no longer a `toColor()` method.  Use the following alternatives:
+
+~~~~
```

Comment:
Does "\`\`\`java" not work in this case? (FYI, Other pages have been using "\`\`\`xtend")